components/match2/wikis/mobilelegends/match_legacy.lua

```diff
@@ -0,0 +1,219 @@
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+	match.links = nil
+
+	if String.isNotEmpty(match.walkover) then
+		match.resulttype = match.walkover
+		match.walkover = match.winner
+	end
+
+	match.staticid = 'Legacy_' .. match2.match2id
+
+	-- Handle extradata fields
+	match.extradata = {}
+	local extradata = Json.parseIfString(match2.extradata)
+	match.extradata.matchsection = extradata.matchsection
+	match.extradata.mvpteam = extradata.mvpteam
+	match.extradata.mvp = extradata.mvp
+	match.extradata.comment = extradata.comment
+
+	local opponents = match2.match2opponents or {}
+	match.extradata.team1icon = (opponents[1] or {}).icon
+	match.extradata.team2icon = (opponents[2] or {}).icon
+	match.extradata.opponent1literal = tostring((opponents[1] or {}).type == 'literal')
+	match.extradata.opponent2literal = tostring((opponents[2] or {}).type == 'literal')
+	match.extradata.opponent1game = 'false'
+	match.extradata.opponent2game = 'false'
+
+	local games = match2.match2games or {}
+	for key, game in ipairs(games) do
+		if String.isNotEmpty(game.vod) then
+			match.extradata['vodgame' .. key] = game.vod
+		end
+	end
+
+	match.extradata = mw.ext.LiquipediaDB.lpdb_create_json(match.extradata)
+
+	-- Handle Opponents
+	local handleOpponent = function (index)
+		local prefix = 'opponent'..index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == 'team' then
+			match[prefix] = opponent.name
+			match[prefix..'score'] = (tonumber(opponent.score) or 0) >= 0 and opponent.score or 0
+			local opponentplayers = {}
+			for i = 1, 10 do
+				local player = opponentmatch2players[i] or {}
+				opponentplayers['p' .. i] = player.name or ''
+				opponentplayers['p' .. i .. 'flag'] = player.flag or ''
+				opponentplayers['p' .. i .. 'dn'] = player.displayname or ''
+			end
+			match[prefix..'players'] = mw.ext.LiquipediaDB.lpdb_create_json(opponentplayers)
+		elseif opponent.type == 'solo' then
+			local player = opponentmatch2players[1] or {}
+			match[prefix] = player.name
+			match[prefix..'score'] = (tonumber(opponent.score) or 0) >= 0 and opponent.score or 0
+			match[prefix..'flag'] = player.flag
+		elseif opponent.type == 'literal' then
+			match[prefix] = 'TBD'
+		end
+	end
+
+	handleOpponent(1)
+	handleOpponent(2)
+
+	return match
+end
+
+function MatchLegacy.storeGames(match, match2)
+	local games = ''
+	for gameIndex, game2 in ipairs(match2.match2games or {}) do
+		local game = Table.deepCopy(game2)
+
+		-- Extradata
+		local extradata = Json.parseIfString(game2.extradata)
+		game.extradata = {}
+		game.extradata.gamenumber = gameIndex
+		for key, item in pairs(extradata) do
+			local teamIndex, typeIndicator, typeIndex = string.match(key, 'team(%d)(%a+)(%d)')
+			if typeIndicator and typeIndex then
+				local newKey = 't' .. teamIndex .. string.sub(typeIndicator, 1, 1) .. typeIndex
+				game.extradata[newKey] = item
+			end
+		end
```

Comment:
What's the use of this?

Reply:
have no clue, could be a leftover of an unused stuff, this maybe removable in the 2nd update